### PR TITLE
load: ignore context cancel error

### PR DESCRIPTION
**Pull request by asjdf**
### Describe the pull request

I recently noticed the error `session: load: read: get: context canceled` in my sentry panel.

I traced the error into this package and found that it will panic when the user closes the connection.

Actually, there are some fixes applied in [line 188](https://github.com/flamego/session/blob/3a174aa4f5f885d873e71bc1bff66444ce19d38f/session.go#L188) but it is not enough.

Because the context canceled before the next handlers, I use `c.ResponseWriter().WriteHeader(0)` to end this handler chain.

By the way, I am looking forward flamego support the method like `Abort()` in gin to prevents pending handlers from being called.

Link to the issue: n/a

### Checklist

- [x] I agree to follow the [Code of Conduct](https://go.dev/conduct) by submitting this pull request.
- [x] I have read and acknowledged the [Contributing guide](https://github.com/flamego/flamego/blob/main/.github/contributing.md).
- [x] I have added test cases to cover the new code.

### CODE REVIEW

**Comment by asjdf** [comment]
@unknwon sorry to bother you, but do you have time to review this pr?

File: session.go [hunk 1]
```diff
@@ -172,6 +172,10 @@ func Sessioner(opts ...Options) flamego.Handler {
 		sid := opt.ReadIDFunc(c.Request().Request)
 		sess, created, err := mgr.load(c.Request().Request, sid, opt.IDLength)
 		if err != nil {
+			if errors.Cause(err) == context.Canceled {
+				c.ResponseWriter().WriteHeader(0)
```
[the remark this answers is not in the document]
Comment (wuhan005) [reply session.go:176]: cc @unknwon

[the remark this answers is not in the document]
Comment (asjdf) [reply session.go:176]: If after the user closes the connection, the other middlewares should continue to execute, but because the ctx has been canceled when loading, the loaded session data is empty, in this case, the Store in line 190 should not be executed? 
In this case, how do we handle the flash message? Or can we use a new ctx for session loading?

**Comment by unknwon** [comment]
Apologies for the delay, will take a look today!

**Comment by asjdf** [comment]
> > 顺便说一下，我期待 flamego 支持像`Abort()`gin 中的方法，以防止挂起的处理程序被调用。
> 
> 如果我正确理解您的问题，您可能需要查看https://flamego.dev/core-services.html#response-stream了解如何存档。
> 
> TL;DR 一旦写入状态代码，就不会调用后续处理程序。

I know about this feature, and that's why I added `c.ResponseWriter().WriteHeader(0)` to the code.

**Comment by asjdf** [comment]
I think there should be some problem with my modification, if you can explain the correct workflow, I can make a modification.

File: session.go (hunk 1 above)
Comment (unknwon) [inline session.go:176]:
```suggestion
				c.ResponseWriter().WriteHeader(http.StatusUnprocessableEntity)
```

**Comment by unknwon** [comment]
I just committed a suggestion, I think things should work as expected now.

**Review by unknwon — APPROVED** [review]
Thanks for the PR!

**Comment by unknwon** [comment]
I'm gonna merge and release a new patch version. If you have any other suggestions that are not included, we can always work on a followup PR, since this one is purely beneficial.

**Comment by unknwon** [comment]
https://github.com/flamego/session/releases/tag/v1.2.2 has been created for this merge.

**Comment by asjdf** [comment]
Thanks for reviewing. 
This blog seems to be misleading me https://neojos.com/blog/2018/03-17-http%E6%80%BB%E7%BB%93-%E7%8A%B6%E6%80%81%E7%A0%81/#0%E8%B6%85%E6%97%B6%E5%AE%A2%E6%88%B7%E7%AB%AF%E4%B8%BB%E5%8A%A8%E6%96%AD%E5%BC%80%E8%BF%9E%E6%8E%A5.
